src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+
+        for (String name : names) {
+            int amount = 0;
+            for (String field : data) {
+                if ((getDate(field).isAfter(getDate(dateFrom))
+                        || getDate(field).isEqual(getDate(dateFrom)))
+                        && (getDate(field).isBefore(getDate(dateTo))
+                        || getDate(field).isEqual(getDate(dateTo)))
+                        && field.contains(name)) {
+                    amount += getEarnedMoney(field);
+                }
+            }
```

Comment:
```suggestion
            for (String field : data) {
                String[] items = inputData.split(" ");
                if (compareDate(items[0], dateFrom, dateTo) && field.contains(name)) {
                    amount += getEarnedMoney(items[2], items[3]);
                }
            }
```
можна зробити метод compareDate() який буде в середині перевіряти потрібну інформацію